From 881ccc1ba86168c32ec56ce284ef8a640ade5392 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:28:39 -0400
Subject: [drm] nouveau/nouveau_acpi: convert acpi_get_handle() to acpi_has_method()

Message-id: <1411129776-21656-5-git-send-email-prarit@redhat.com>
Patchwork-id: 96340
O-Subject: [RHEL7.1 PATCH BZ 1141846 04/61] nouveau_acpi: convert acpi_get_handle() to acpi_has_method()
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

commit 187b5b5d520c2318a1f88fb8d8913a9d7fbf7d92
Author: Zhang Rui <rui.zhang@intel.com>
Date:   Tue Sep 3 08:32:00 2013 +0800

    nouveau_acpi: convert acpi_get_handle() to acpi_has_method()

    acpi_has_method() is a new ACPI API introduced to check
    the existence of an ACPI control method.

    It can be used to replace acpi_get_handle() in the case that
    1. the calling function doesn't need the ACPI handle of the control method.
    and
    2. the calling function doesn't care the reason why the method is unavailable.

    Convert acpi_get_handle() to acpi_has_method()
    in drivers/gpu/drm/nouveau/nouveau_acpi.c in this patch.

    Signed-off-by: Zhang Rui <rui.zhang@intel.com>
    CC: Ben Skeggs <bskeggs@redhat.com>
    CC: Dave Airlie <airlied@redhat.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: David Airlie <airlied@redhat.com>
Cc: Adam Jackson <ajax@redhat.com>
Cc: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/nouveau/nouveau_acpi.c b/drivers/gpu/drm/nouveau/nouveau_acpi.c
index 8335674..53942c0 100644
--- a/drivers/gpu/drm/nouveau/nouveau_acpi.c
+++ b/drivers/gpu/drm/nouveau/nouveau_acpi.c
@@ -253,18 +253,15 @@ static struct vga_switcheroo_handler nouveau_dsm_handler = {
 
 static int nouveau_dsm_pci_probe(struct pci_dev *pdev)
 {
- acpi_handle dhandle, nvidia_handle;
- acpi_status status;
+ acpi_handle dhandle;
  int retval = 0;
 
  dhandle = DEVICE_ACPI_HANDLE(&pdev->dev);
  if (!dhandle)
   return false;
 
- status = acpi_get_handle(dhandle, "_DSM", &nvidia_handle);
- if (ACPI_FAILURE(status)) {
+ if (!acpi_has_method(dhandle, "_DSM"))
   return false;
- }
 
  if (nouveau_test_dsm(dhandle, nouveau_dsm, NOUVEAU_DSM_POWER))
   retval |= NOUVEAU_DSM_HAS_MUX;
-- 
1.7.1